src/commands/env/api.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { readFile } from 'node:fs/promises';
+import got, { Headers, Method } from 'got';
+import * as chalk from 'chalk';
+import * as ProxyAgent from 'proxy-agent';
+import { getProxyForUrl } from 'proxy-from-env';
+import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
+import { SfError, Messages, Org } from '@salesforce/core';
+import { CliUx } from '@oclif/core';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.load('@cristiand391/sf-plugin-api', 'env.api', [
+  'summary',
+  'description',
+  'examples',
+  'flags.target-org.summary',
+  'flags.method.summary',
+  'flags.include.summary',
+  'flags.header.summary',
+  'flags.body.summary',
+  'errors.invalid-http-header',
+]);
+
+export default class EnvApi extends SfCommand<string> {
+  public static summary = messages.getMessage('summary');
+  public static description = messages.getMessage('description');
+  public static examples = messages.getMessages('examples');
+  public static enableJsonFlag = false;
+  public static flags = {
+    'target-org': Flags.requiredOrg({
+      summary: messages.getMessage('flags.target-org.summary'),
+      char: 'o',
+    }),
+    include: Flags.boolean({
+      summary: messages.getMessage('flags.include.summary'),
+      default: false,
+    }),
+    method: Flags.enum<Method>({
+      options: ['GET', 'POST', 'PUT', 'PATCH', 'HEAD', 'DELETE', 'OPTIONS', 'TRACE'],
+      summary: messages.getMessage('flags.method.summary'),
+      char: 'X',
+      default: 'GET',
+    }),
+    header: Flags.string({
+      summary: messages.getMessage('flags.header.summary'),
+      char: 'H',
+      multiple: true,
+    }),
+    body: Flags.file({
+      summary: messages.getMessage('flags.body.summary'),
+    }),
+  };
+
+  public static args = [
+    {
+      name: 'endpoint',
+      description: 'Salesforce API endpoint.',
+      required: true,
+    },
+  ];
+
+  private static getHeaders(keyValPair: string[]): Headers {
+    const headers = {};
+
+    for (const header of keyValPair) {
+      const split = header.split(':');
+      if (split.length !== 2) {
+        throw new SfError(messages.getMessage('errors.invalid-http-header', [header]), '', [
+          'Make sure the header is in a "key:value" format, e.g. "Accept: application/json"',
+        ]);
+      }
+      headers[split[0]] = split[1];
+    }
+
+    return headers;
+  }
+
+  public async run(): Promise<string> {
+    const { flags, args } = await this.parse(EnvApi);
+
+    let body: Buffer;
+
+    if (flags.body) {
+      body = await readFile(flags.body);
```

Comment:
If `body` is passed, it would be convenient if the appropriate `Content-Type` header was added for you automatically.

Reply:
Good idea, the command pass the body file to got as a `Buffer` so some headers are set:
https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#body

to get the content-type of the buffer we could use this
https://github.com/sindresorhus/file-type/

I'll add a new issue for this 👍🏼